scripts/make_api_rst.py

```diff
@@ -36,6 +36,55 @@
 Welcome to the QGIS Python API documentation project
 ==============================================================
 
+Introduction
+------------
+
+`QGIS <https://qgis.org>`_ is a user friendly Open Source Geographic Information System (GIS) that runs on Linux, Unix, Mac OSX, and Windows.
+QGIS supports vector, raster, and database formats. QGIS is licensed under the GNU General Public License.
+QGIS lets you browse and create map data on your computer. It supports many common spatial data formats (e.g. ESRI ShapeFile, GeoPackage, geotiff).
+QGIS supports plugins to do things like display tracks from your GPS. QGIS is Open Source software and its free of cost (download here).
+We welcome contributions from our user community in the form of code contributions, bug fixes, bug reports, contributed documentation,
+advocacy and supporting other users on our mailing lists and forums. Financial contributions are also welcome.
+
+QGIS source code is available at https://github.com/qgis/QGIS.
+There is also a `C++ version of the API documentation <https://api.qgis.org>`_ available.
+
+Earlier versions of the API
+---------------------------
+
+See `Backwards Incompatible Changes <https://api.qgis.org/api/master/api_break.html>`_ for information about incompatible changes to API between releases.
+
+Earlier versions of the documentation are also available on the QGIS website:
+`3.30 <https://qgis.org/pyqgis/3.30>`_,
+`3.28 (LTR) <https://qgis.org/pyqgis/3.28>`_,
+`3.26 <https://qgis.org/pyqgis/3.26>`_,
+`3.24 <https://qgis.org/pyqgis/3.24>`_,
+`3.22 (LTR) <https://qgis.org/pyqgis/3.22>`_,
+`3.20 <https://qgis.org/pyqgis/3.20>`_,
+`3.18 <https://qgis.org/pyqgis/3.18>`_,
+`3.16 (LTR) <https://qgis.org/pyqgis/3.16>`_,
+`3.14 <https://qgis.org/pyqgis/3.14>`_,
+`3.12 <https://qgis.org/pyqgis/3.12>`_,
+`3.10 (LTR) <https://qgis.org/pyqgis/3.10>`_,
+`3.8 <https://qgis.org/pyqgis/3.8>`_,
+`3.6 <https://qgis.org/pyqgis/3.6>`_,
+`3.4(LTR) <https://qgis.org/pyqgis/3.4>`_,
+`3.2 <https://qgis.org/pyqgis/3.2>`_,
+`3.0 <https://qgis.org/pyqgis/3.0>`_,
+
```

Comment:
This is going something to maintain. Do we really want this?

Reply:
> This is going something to maintain. Do we really want this?

I think we can kill the [index page](https://qgis.org/pyqgis) that we manually update today. It wouldn't make sense to keep duplicate info, and it will leave us with same level of manual maintenance.
We can go further, with an automatic process as done in the code repo to fill the same list in C++ API doc (https://github.com/qgis/QGIS/blob/ba48826ca8506907362111377508b95a8e99c898/scripts/release.pl#L228); elementary information are in pyqgis_conf.yml. How-to is a bit beyond my skills...
https://github.com/qgis/pyqgis/blob/d7528a4e2d7171b7cbcb44cf0c7ae1d5baa25b6f/pyqgis_conf.yml#L2-L3